java-checks/src/main/java/org/sonar/java/checks/ServletMethodsExceptionsThrownCheck.java

```diff
@@ -104,20 +104,43 @@ private void checkMethodInvocation(MethodInvocationTree node) {
     if (symbol.isMethodSymbol()) {
       List<Type> types = ((Symbol.MethodSymbol) symbol).thrownTypes();
       if (!types.isEmpty()) {
-        addIssueIfNotCatched(types, ExpressionUtils.methodName(node), "Add a \"try/catch\" block for \"" + symbol.name() + "\".");
+        addIssueIfNotCaught(types, ExpressionUtils.methodName(node), symbol.name());
       }
     }
   }
 
-  private void addIssueIfNotCatched(Iterable<Type> thrown, Tree node, String message) {
+  private void addIssueIfNotCaught(Type thrown, Tree node) {
+    if (isNotCaught(thrown)) {
+      String message = "Handle the " + "\"" + thrown + "\"" + " thrown here in a \"try/catch\" block.";
+      reportIssue(node, message);
+    }
+  }
+
+  private void addIssueIfNotCaught(Iterable<Type> thrown, Tree node, String symbolName) {
+    List<Type> uncaughtTypes = new ArrayList<>();
     for (Type type : thrown) {
-      if (isNotcatched(type)) {
-        reportIssue(node, message);
+      if (isNotCaught(type)) {
+        uncaughtTypes.add(type);
+      }
+    }
+    if (!uncaughtTypes.isEmpty()) {
+      reportIssue(node, buildMessage(symbolName, uncaughtTypes));
+    }
+  }
+
+  private static String buildMessage(String symbolName, List<Type> uncaughtTypes) {
+    StringBuilder stringBuilder = uncaughtTypes.size() == 1 ? new StringBuilder("Handle the following exception that could be thrown by " + "\"" + symbolName + "\": ") :
+      new StringBuilder("Handle the following exceptions that could be thrown by " + "\"" + symbolName + "\": ");
+    for (int i = 0; i < uncaughtTypes.size(); i++) {
```

Comment:
Instead of relying on a string builder, you can directly stream it from the initial list:
```
uncaughtTypes.stream().map(Type::name).collect(Collectors.join(", "));
```